src/libraries/System.Runtime/tests/System/UIntPtrTests.GenericMath.cs

```diff
@@ -2223,7 +2224,7 @@ public static void CreateSaturatingFromDoubleTest()
             Assert.Equal(nuint.MaxValue, NumberBaseHelper<nuint>.CreateSaturating<double>(double.MaxValue));
             Assert.Equal(nuint.MinValue, NumberBaseHelper<nuint>.CreateSaturating<double>(double.MinValue));
 
-            Assert.Equal(nuint.MinValue, NumberBaseHelper<nuint>.CreateSaturating<double>(double.NaN));
+            Assert.Equal((arch == Architecture.X86 || arch == Architecture.X64)?nuint.MaxValue:nuint.MinValue, NumberBaseHelper<nuint>.CreateSaturating<double>(double.NaN));
```

Comment:
Makes sense. Changed the implementation. Handled the NaN cases explicitly in the TryConvertTo instead of using Architecture to differentiate between outputs.